executor/src/constant_evaluator/mod.rs

```diff
@@ -35,45 +42,46 @@ fn generate_values<T: FieldElement>(
         computed_columns,
     };
     // TODO we should maybe pre-compute some symbols here.
-    match body {
+    let result = match body {
         FunctionValueDefinition::Expression(e) => (0..degree)
             .into_par_iter()
             .map(|i| {
                 // We could try to avoid the first evaluation to be run for each iteration,
                 // but the data is not thread-safe.
                 let fun = evaluator::evaluate(e, &symbols).unwrap();
                 evaluator::evaluate_function_call(fun, vec![Rc::new(T::from(i).into())], &symbols)
-                    .unwrap()
-                    .try_to_number()
-                    .unwrap()
+                    .and_then(|v| v.try_to_number())
             })
-            .collect(),
-        FunctionValueDefinition::Array(values) => {
-            let values: Vec<_> = values
-                .iter()
-                .flat_map(|elements| {
-                    let items = elements
-                        .pattern()
-                        .iter()
-                        .map(|v| {
-                            evaluator::evaluate(v, &symbols)
-                                .unwrap()
-                                .try_to_number()
-                                .unwrap()
-                        })
-                        .collect::<Vec<_>>();
+            .collect::<Result<Vec<_>, _>>(),
+        FunctionValueDefinition::Array(values) => values
+            .iter()
+            .map(|elements| {
+                let items = elements
+                    .pattern()
+                    .iter()
+                    .map(|v| evaluator::evaluate(v, &symbols).and_then(|v| v.try_to_number()))
+                    .collect::<Result<Vec<_>, _>>()?;
 
-                    items
-                        .into_iter()
-                        .cycle()
-                        .take(elements.size() as usize)
-                        .collect::<Vec<_>>()
-                })
-                .collect();
-            assert_eq!(values.len(), degree as usize);
-            values
-        }
+                Ok(items
+                    .into_iter()
+                    .cycle()
+                    .take(elements.size() as usize)
+                    .collect::<Vec<_>>())
+            })
+            .collect::<Result<Vec<_>, _>>()
+            .map(|values| {
+                let values: Vec<T> = values.into_iter().flatten().collect();
+                assert_eq!(values.len(), degree as usize);
+                values
+            }),
         FunctionValueDefinition::Query(_) => panic!("Query used for fixed column."),
+    };
+    match result {
+        Err(err) => {
+            eprintln!("Error evaluating fixed polynomial {name} = {body}:\n{err}");
```

Comment:
Is it fine to print to stderr here or should we propagate the error further upwards?
The problem with propagating is that we have to create yet another error class that contains the name and the body - or just turn it into a string here.

Reply:
Is there a possibility for the error to be handled somewhere else later (e.g., nicer reporting maybe), so that the temporary unwraps elsewhere are worth it? If so maybe `Result<_, String>` is enough?

Reply:
I think eventually we should really refactor all the error stuff to properly propagate errors upwards. I think the `backend` crate already does some of that.